### PR TITLE
Use photo framework instead assets library

### LINKED ISSUE #667: When selecting to upload an image within the ownCloud app not all the images available on the device are shown
#- Starting point ( 😟 ) we have not been able to reproduce this, after having tried on several devices
- STEPS:
  + Check the number of images available on the device from the photo app
  + Select to upload an image from the ownCloud app, check that the number is not the same, less images are shown
  + However, if you access to the photp app and select to upload to ownCloud any of the non-shown images/videos, it's uploaded just fine
   + Note, all images are shown just fine on other apps
   + Note2, it was reporting on an iPhone6 iOS9.3.1

- To try to fix this we will modify what files are shown within the upload view to show everything, currently, we are not showing all the folders available on the photo app

- [x] [GIT] Create branch "use_photo_framework_insetad_assets_library" @javiergonzper 
- [X] [GIT] Merge branch "use_photo_framework_insetad_assets_library" @nasli
- [X] [QA] Create Test plan [not needed]
- [X] [QA] Validate Test plan
- [X] [DEV] Update the ELCAlbumPicker with this fork:https://github.com/amitbobade2007/ELCImagePickerController @javiergonzper
- [X] [DEV] Modify the ELCAlbumPicker to get all the albums @javiergonzper
- [X] [DEV] Modify the library to temove the IS_IOS_8 to avoid conflicts with out IS_IOS_8 @javiergonzper
- [X] [DEV] Add the Select folder button @javiergonzper
- [X] [DEV] Center the images @javiergonzper
- [X] [DEV] Add the overlay with the duration of the videos @javiergonzper
- [X] [DEV] Send the seleced items to be uplaoded @javiergonzper
- [X] [DEV] Fix the internationalization problems @javiergonzper
- [X] [DEV] Create a method to get the filename of the image or video @nasli
- [X] [DEV] Check style title navigation bar @mRoRo

PR: #679

### PR DESCRIPTION
**Pull request by nasli**
This closes #667

### CODE REVIEW

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAlbumPickerController.m [hunk 1]
```diff
@@ -17,196 +20,307 @@
...
-    
-    cell.textLabel.text = [NSString stringWithFormat:@"%@ (%ld)",[g valueForProperty:ALAssetsGroupPropertyName], (long)gCount];
-    [cell.imageView setImage:[UIImage imageWithCGImage:[(ALAssetsGroup*)[self.assetGroups objectAtIndex:indexPath.row] posterImage]]];
-	[cell setAccessoryType:UITableViewCellAccessoryDisclosureIndicator];
-	
+    // Increment the cell's tag
+    NSInteger currentTag = cell.tag + 1;
+    cell.tag = currentTag;
+    
+//    if(indexPath.row == 0) {
```
Comment (mRoRo) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAlbumPickerController.m]: Delete the commented code if it's not usefull

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m [hunk 2]
```diff
@@ -7,84 +7,105 @@
...
+            PHAsset *phAsset = (PHAsset *)asset.asset;
+            [self.imageManager requestImageForAsset:phAsset targetSize:CGSizeMake(70, 70) contentMode:PHImageContentModeAspectFill options:options resultHandler:^(UIImage * result, NSDictionary * info) {
+                imageView.image = result;
+            }];
+            
         } else {
-            UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageWithCGImage:asset.asset.thumbnail]];
+            UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 70, 70)];
+            PHAsset *phAsset = (PHAsset *)asset.asset;
+            [self.imageManager requestImageForAsset:phAsset targetSize:CGSizeMake(70, 70) contentMode:PHImageContentModeAspectFill options:options resultHandler:^(UIImage * result, NSDictionary * info) {
```
Comment (mRoRo) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m]: This code is also in the else structure.

 [self.imageManager requestImageForAsset:phAsset targetSize:CGSizeMake(70, 70) contentMode:PHImageContentModeAspectFill options:options resultHandler:^(UIImage * result, NSDictionary * info) {                imageView.image = result;
            }];

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m [hunk 3]
```diff
@@ -7,84 +7,105 @@
...
-            [overlayView release];
+            overlayView.labIndex.text = [NSString stringWithFormat:@"%d", asset.index + 1];
         }
     }
 }
 
 - (void)cellTapped:(UITapGestureRecognizer *)tapRecognizer
 {
     CGPoint point = [tapRecognizer locationInView:self];
-    //CGFloat totalWidth = self.rowAssets.count * 75 + (self.rowAssets.count - 1) * 4;
```
Comment (mRoRo) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m]: Remove if it's not usefull

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetTablePicker.m [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        
+        
+        NSIndexPath *ip = [NSIndexPath indexPathForRow:_rowToScrollAfterRotation
+                                             inSection:section];
+        [self.tableView scrollToRowAtIndexPath:ip
+                              atScrollPosition:UITableViewScrollPositionBottom
+                                      animated:NO];
+    }
+    
+    //[self.navigationItem setTitle:[NSString stringWithFormat:@"%@",[_assetGroup valueForProperty:ALAssetsGroupPropertyName]]];
```
Comment (mRoRo) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetTablePicker.m]: Remove if it's unusefull

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCImagePickerController.m [hunk 5]
```diff
@@ -6,34 +6,83 @@
...
 
-- (void) viewWillAppear:(BOOL)animated {
-    [super viewWillAppear:animated];
-    
-    AppDelegate *app = (AppDelegate *)[[UIApplication sharedApplication] delegate];
-    app.currentViewVisible = self;
+- (id)initImagePicker
+{
+    ELCAlbumPickerController *albumPicker = [[ELCAlbumPickerController alloc] init];
+    //ELCAlbumPickerController *albumPicker = [[ELCAlbumPickerController alloc] initWithStyle:UITableViewStylePlain];
```
Comment (mRoRo) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCImagePickerController.m]: Remove the comment if not usefull

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCOverlayImageView.m [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        
+        if ([[ELCConsole mainConsole] onOrder]) {
+            self.labIndex = [[UILabel alloc] initWithFrame:CGRectMake(5, 5, 16, 16)];
+            self.labIndex.backgroundColor = [UIColor redColor];
+            self.labIndex.clipsToBounds = YES;
+            self.labIndex.textAlignment = NSTextAlignmentCenter;
+            self.labIndex.textColor = [UIColor whiteColor];
+            self.labIndex.layer.cornerRadius = 8;
+            self.labIndex.layer.shouldRasterize = YES;
+            //        self.labIndex.layer.borderWidth = 1;
```
Comment (mRoRo) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCOverlayImageView.m]: Remove comments if not usefull

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m (hunk 2 above)
Comment (nasli) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetCell.m]: The option could be a new method but we will need to pass all the parameters and will be a similar method that requestImageForAsset, I rather keep it simpler as it is

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetTablePicker.m (hunk 4 above)
Comment (nasli) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCAssetTablePicker.m]: 👍

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCImagePickerController.m (hunk 5 above)
Comment (nasli) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCImagePickerController.m]: 👍

File: Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCOverlayImageView.m (hunk 6 above)
Comment (nasli) [inline Owncloud iOs Client/ExternalLibreries/ELCPickerLib/ELCOverlayImageView.m]: 👍

**Pull request merged**